packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1390,12 +1410,12 @@ abstract class FlutterCommand extends Command<void> {
   ///
   /// Returns a record of key and value as strings.
   MapEntry<String, String> _parseProperty(String line) {
-    final RegExp blockRegExp = RegExp(r'^\s*([a-zA-Z_]+[a-zA-Z0-9_]*)\s*=\s*"""\s*(.*)$');
+    final RegExp blockRegExp = RegExp(DotEnvRegex.block);
```

Comment:
I believe what Zach meant was lifting these invocations of hte `RegExp` constructor out of these methods and into static fields that are only allocated once. I believe the string literals are already constant.

Reply:
yes, they should be final. the point here is not about making them constant, but about avoiding a new allocation each time the methods are invoked.

Reply:
I moved all the `RegExp` constructors. They are now build only one time.